protocol/tx_test.go

```diff
@@ -1,4 +1,4 @@
-package protocol
```

Comment:
IMO, the separate memstore package is cleaner than using `_test` packages. What about moving the memstore package to `prottest/memstore`?